src/axom/sidre/core/View.hpp

```diff
@@ -1514,10 +1547,85 @@ class View
   State getStateId(const std::string& name) const;
 
   /*!
-   * \brief Private method. If allocatorID is a valid allocator ID then return
-   *  it. Otherwise return the ID of the default allocator of the owning group.
+   * \brief Private method. If allocatorID is a valid Axom allocator ID then return
+   *  it. Otherwise return the ID of the default Axom allocator of the owning group.
    */
-  int getValidAllocatorID(int allocatorID);
+  int getValidAxomAllocatorID(int allocatorID);
+
+  /*!
+   * \brief Private method. If allocatorID is a valid Axom allocator
+   *  ID then return its corresponding Conduit allocator ID. Otherwise
+   *  return the corresponding Conduit allocator ID of the default
+   *  allocator of the owning group.
+   */
+  int getValidConduitAllocatorID(int allocatorID);
+
+  /*!
+   * \brief Return whether view data is accessible on the host CPU,
+   * as determined by Axom's memory management.
+   *
+   * If data is external and not allocated through Umpire or malloc,
+   * the behavior is undefined.
+   *
+   * By convention, this returns a false if data is not allocated,
+   * because we expect null pointers to be correctly checked before use.
+   */
+  bool isHostAccessible() const;
+
+  //!@brief Print on host, as a single line.
+  template <typename T>
+  void hostPrintScalar(std::ostream& os = std::cout) const
+  {
+    if(isHostAccessible())
+    {
+      os << ' ' << T(m_node.value());
+    }
+    else
+    {
+      os << ' ' << getVoidPtr() << " # non-host " << typeid(T).name() << " data";
+    }
+  }
+
+  //!@brief Print on host, as a single line.
+  template <typename T>
+  void hostPrintArray(std::ostream& os = std::cout) const
```

Comment:
Array isn't really a term associated with a Sidre View.  What about just a generic hostPrint() that supports both the scalar and your array case?

Reply:
There is a `hostPrint` that does that.  It delegates to template methods `hostPrintScalar` and `hostPrintArray` depending on the `View` state.  I think I made it confusing because I put those two methods in the public section.  I'll move them to private, where they belong.  Will that do?